libero-config/bin/ConfigDistributor.js

```diff
@@ -0,0 +1,127 @@
+        jsonFileName: '/source/js/derived-from-config/configForJs.json'
+      }
+    };
+  }
+
+  distribute(configPaths, configGenerator) {
+
+    console.log('Distributing config...');
+
+    return configGenerator.generateConfig(configPaths)
+
+                          .then((config) => {
+                            return Promise.all(
+                              [
+                                this.distributeToSass(config.layerAllocations.sass, config.data),
+                                this.distributeToJs(config.layerAllocations.js, config.data),
+                              ]
+                            )
+                          })
+
+                          .catch(err => {
+                            console.error(err.message);
+                            process.exit(1);
+                          });
+  }
+
+  distributeToJs(allocations, data) {
+    return ConfigDistributor.writeFile(
+      ConfigDistributor.processForJs(allocations, data),
+      this.paths.out.jsonFileName
+    );
+  }
+
+  distributeToSass(allocations, data) {
+
+    const fileWritePromises = [];
+
+    // Each allocation is written to a separate file
+    allocations.forEach((allocation) => {
+      const dataForAllocation = {};
+      dataForAllocation[allocation] = data[allocation];
+      const processedItemData = ConfigDistributor.processForSass(dataForAllocation);
+      const outFileName =
+        `${this.paths.out.sassVariablesFileNameRoot}${allocation}.scss`;
+      fileWritePromises.push(
+        new Promise((resolve) => {
+          resolve(ConfigDistributor.writeFile(processedItemData, outFileName));
+        })
+      );
+    });
+
+    return Promise.all(fileWritePromises).catch(err => { throw err; } );
+
+  }
+
+  static processForJs(allocations, data) {
+    const processed = {};
+    allocations.forEach((allocation) => {
+      processed[allocation] = data[allocation];
+    });
+    return JSON.stringify(processed);
+  }
+
+  static processForSass(data) {
+    const deepData = deepIterator(data);
+    for (let {parent, key, value} of deepData) {
+      if (value instanceof Color) {
+        parent[key] = value.rgb().string();
+      }
+    }
+
+    return Object.entries(flatten(data, {delimiter: '-'}))
+                 .reduce((carry, pair) => {
+                   const [key, value] = pair;
+                   return `${carry}$${key}: ${value};\n`;
+                 }, '');
+  }
+
+  static writeDirectory(path) {
+    return new Promise((resolve, reject) => {
+      fs.mkdir(path, { recursive: true}, (err) => {
+        if (err) {
+          reject(err);
+        }
+        resolve();
+      });
+    });
+  }
+
+  static async writeFile(data, outPath) {
+    let projectRootPath = process.cwd();
+    const matched = projectRootPath.match(/^.*\/libero-config\/bin.*$/);
+    if (matched) {
+      projectRootPath = path.resolve(path.join(process.cwd(), '../..'));
```

Comment:
`../..` may not work on Windows systems
```suggestion
      projectRootPath = path.resolve(path.join(process.cwd(), '..', '..'));
```